frontend/apps/web/components/RunTImeline/RunTimeline.tsx

```diff
@@ -0,0 +1,457 @@
+    const { formatDate, getPositionPercentage, timeLabels } = props;
+
+    return (
+      <div className="w-full sticky top-0 h-14 border-b border-gray-200 dark:border-gray-700 bg-gray-200 dark:bg-gray-800 z-10 ">
+        <div className="relative w-full h-full">
+          {timeLabels.map((label, index) => (
+            <div
+              key={index}
+              className="absolute top-0 text-xs text-gray-700 dark:text-gray-300"
+              style={{ left: `${getPositionPercentage(label)}%` }}
+            >
+              <div className="whitespace-nowrap py-1">{formatDate(label)}</div>
+              <div className="whitespace-nowrap">{formatTime(label)}</div>
+              <div className="h-4 w-[1px] rounded-full bg-gray-500 mx-auto" />
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  }
+}
+
+function convertTimestampToDate(
+  timestamp: { seconds: bigint; nanos: number } | undefined
+): Date {
+  if (!timestamp) return new Date();
+
+  const millisecondsFromSeconds = Number(timestamp.seconds) * 1000;
+  const millisecondsFromNanos = timestamp.nanos / 1_000_000;
+
+  const totalMilliseconds = millisecondsFromSeconds + millisecondsFromNanos;
+
+  return new Date(totalMilliseconds);
+}
+
+function getCloseOrErrorOrCancelDate(task: JobRunEvent): Date {
+  const errorTask = task.tasks.find((item) => item.error);
+  const errorTime = errorTask ? errorTask.eventTime : undefined;
+  const cancelTime = task.tasks.find(
+    (t) => t.type === 'ActivityTaskCancelRequested'
+  )?.eventTime;
+  return errorTime
+    ? convertTimestampToDate(errorTime)
+    : cancelTime
+      ? convertTimestampToDate(cancelTime)
+      : convertTimestampToDate(task.closeTime);
+}
+
+interface SyncLabelProps {
+  task: JobRunEvent;
+}
+
+function SyncLabel(props: SyncLabelProps) {
+  const { task } = props;
+
+  const schemaTable = `${task.metadata?.metadata.value?.schema}.${task.metadata?.metadata.value?.table} `;
+
+  return (
+    <div className="text-white">
+      {task.metadata?.metadata.case == 'syncMetadata' && schemaTable}
+    </div>
+  );
+}
+
+interface ActivityLabelProps {
+  task: JobRunEvent;
+  getTaskStatus: (task: JobRunEvent) => string;
+}
+
+function ActivityLabel({ task, getTaskStatus }: ActivityLabelProps) {
+  const status = getTaskStatus(task);
+
+  const handleStatus = () => {
+    switch (status) {
+      case 'completed':
+        return <CheckCircledIcon className="text-green-500" />;
+      case 'failed':
+        return <CrossCircledIcon className="text-red-500" />;
+      case 'canceled':
+        return <MinusCircledIcon className="text-yellow-500" />;
+      case 'running':
+        return <Spinner />;
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <div className="flex flex-row items-center gap-2">
+      {task.id.toString()}.
+      <TruncatedText text={task.type} />
+      {handleStatus()}
+    </div>
+  );
+}
+
+interface StatusFilterProps {
+  selectedStatuses: string[];
```

Comment:
Would it be difficult to narrow the status type to be not a string, but of the types that you have defined throughout this file? completed, failed, running, canceled?
Will make this more typesafe and readable throughout this file.

Reply:
Yeah definitely need more types - just hadn't gotten to it yet. Updated now, lmk what you think.